test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -643,6 +750,52 @@ public void Start<TContext>(IHttpApplication<TContext> application)
             }
         }
 
+        internal class StartupVerifyServiceA : IStartup
+        {
+            internal ServiceA ServiceA { get; set; }
+
+            internal ServiceDescriptor ServiceADescriptor { get; set; }
+
+            public IServiceProvider ConfigureServices(IServiceCollection services)
+            {
+                ServiceADescriptor = services.FirstOrDefault(s => s.ServiceType == typeof(ServiceA));
+
+                return services.BuildServiceProvider();
+            }
+
+            public void Configure(IApplicationBuilder app)
+            {
+                ServiceA = app.ApplicationServices.GetService<ServiceA>();
+            }
+        }
+
+        public class TestHostingStartup : IHostingStartup
+        {
+            public void Configure(IWebHostBuilder builder)
+            {
+                var loggerProvider = new TestLoggerProvider();
+                builder.UseSetting("testhostingstartup", "1")
+                       .ConfigureServices(services => services.AddSingleton<ServiceA>())
+                       .ConfigureServices(services => services.AddSingleton<ITestSink>(loggerProvider.Sink))
+                       .ConfigureLogging(lf => lf.AddProvider(loggerProvider));
+            }
+        }
+
+        public class TestLoggerProvider : ILoggerProvider
```

Comment:
Anything in https://github.com/aspnet/Logging/tree/dev/src/Microsoft.Extensions.Logging.Testing you can use?